osu.Game/Skinning/SkinInfo.cs

```diff
@@ -39,6 +39,21 @@ public class SkinInfo : RealmObject, IHasRealmFiles, IEquatable<SkinInfo>, IHasG
 
         public bool Protected { get; set; }
 
+        /// <summary>
+        /// The latest version in YYYYMMDD format for skin layout migrations.
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>20240625: Moves combo counters from ruleset-agnostic to ruleset-specific HUD targets.</description></item>
+        /// </list>
+        /// </remarks>
+        public const int LATEST_LAYOUT_VERSION = 20240625;
```

Comment:
Not sure how to feel about having to maintain yet another versioning scheme in general. Can this not be done in a one-and-done manner?

Reply:
Paraphrased from feedback on discord:

> from an avoiding-over-engineering perspective using a realm migration one-and-done operation sounds okay, but one thing to keep in mind that anyone who has shared or uploaded a lazer skin online with the modifications will be broken. i'm not sure that's such a common thing, but worth noting.
>
> adding the skin version does sound like the more correct way forward. i understand spaceman's opinion on the review comment but don't really see a way forward without versioning of some kind (we're always going to need a way to make changes to the schema).
>
> that said, currently `LayoutVersion` doesn't save to the skin json. it would need to in order to be useful (ie. if a user imports a skin which was exported from a previous version).

In addition, I really feel like the migration code should be isolated to a single method somewhere if possible. I haven't yet assessed if it can *not* exist in two classes as per this PR, but even if it is, I'd want any and all migration code to be in a `private void runMigrations()` method (or two).

Reply:
Applied in https://github.com/ppy/osu/pull/26249/commits/0c34e7bebbce5a989af00d8daed25331ff33321a, I've moved the migration logic to a single method in `Skin`.